Android/src/org/droidplanner/android/proxy/mission/item/fragments/MissionWaypointFragment.java

```diff
@@ -45,9 +45,11 @@ public void onApiConnected() {
         altitudePicker.setViewAdapter(altitudeAdapter);
         altitudePicker.addScrollListener(this);
 
-        final Waypoint item = (Waypoint) getMissionItems().get(0);
-        delayPicker.setCurrentValue((int) item.getDelay());
-        altitudePicker.setCurrentValue(lengthUP.boxBaseValueToTarget(item.getCoordinate().getAltitude()));
+        if(getMissionItems().size() > 0) {
```

Comment:
@droneboost The mission items list is expected to never be empty, since that's the requirement for displaying this fragment. The check is fine, but let's enforce the requirement by throwing an exception if the list is empty.

That should allow us to detect error conditions as well.